pkgs/applications/networking/browsers/firefox/wrapper.nix

```diff
@@ -81,6 +92,61 @@ let
             ++ pkcs11Modules;
       gtk_modules = [ libcanberra-gtk2 ];
 
+      #########################
+      #                       #
+      #   EXTRA PREF CHANGES  #
+      #                       #
+      #########################
+      policiesJson = builtins.toFile "policies.json"
+        (builtins.toJSON enterprisePolicies);
+
+      extensions = builtins.map (a:
+        if ! (builtins.hasAttr "extid" a) then
+        throw "extraExtensions has an invalid entry. Missing extid attribute. Please use fetchfirefoxaddon"
+        else
+        a
+      ) extraExtensions;
+
+      enterprisePolicies =
+      {
+        policies = {
+          DisableAppUpdate = true;
+        } //
+        {
+          ExtensionSettings = {
+            "*" = {
+                blocked_install_message = "You can't have manual extension mixed with nix extensions";
+                installation_mode = "blocked";
+              };
+
+          } // lib.foldr (e: ret:
+              ret // {
+                "${e.extid}" = {
+                  installation_mode = "allowed";
+                };
+              }
+            ) {} extensions;
+        }
+        // extraPolicies;
+      };
+
+      mozillaCfg = builtins.toFile "mozilla.cfg" ''
+// First line must be a comment
```

Comment:
This looks indented incorrectly.

Reply:
Its not because toFile does not strip tabs away and this file format needs a comment without spaces at the beginning

Reply:
@Luis-Hebendanz: `toFile` only takes strings, the stripping is caused by `''...''`. I don't however see any tab characters involved here. Checking this quickly via `nix repl` results in:

```
nix-repl> builtins.readFile (builtins.toFile "mozilla.cfg" ''
                  // First line must be a comment
          
                  // Disables addon signature checking
                  // to be able to install addons that do not have an extid
                  // Security is maintained because only user whitelisted addons
                  // with a checksum can be installed
                  lockPref("xpinstall.signatures.required", false);
                  ...
          '')
"// First line must be a comment\n\n// Disables addon signature checking\n// to be able to install addons that do not have an extid\n// Security is maintained because only user whitelisted addons\n// with a checksum can be installed\nlockPref(\"xpinstall.signatures.required\", false);\n...\n"

nix-repl>
```